### PR TITLE
docs: update readme and contributing files

### LINKED ISSUE #164: Update Contributors Guide to Expand on Frontend Conventions
Componsition API, data derivations, etc....

### PR DESCRIPTION
**Pull request by mds1**
Resolves #164

### CODE REVIEW

File: CONTRIBUTING.md
Comment (apbendi) [inline CONTRIBUTING.md:165]: Wondering out loud if we should change the title of this section to "Frontend Contribution Guidelines" for extra clarity. There's some ambiguity about whether a web3 "app" means just the frontend or frontend + contracts.

File: CONTRIBUTING.md
Comment (apbendi) [inline CONTRIBUTING.md:182]: Suggested simplification for the first sentence: "This is ok for simple components."

**Review by apbendi — APPROVED** [review]
Left some very minor comments, but looks great to me! I'll leave it to @thelostone-mc and @gdixon to weigh in the substance, since I'm not a savvy enough frontend dev to have strong opinions.

File: CONTRIBUTING.md
Comment (mds1) [reply CONTRIBUTING.md:165]: Should we rename the root folder from `app` to `frontend` then? I went with app here so they'd be consistent

Comment (apbendi) [reply CONTRIBUTING.md:165]: I think it's ok to leave the folder named `app`, but for someone not yet familiar with that naming convention, using Frontend here might be clearer. Don't feel strongly about it, happy to hear what others think and fine either way.

**Review by thelostone-mc — APPROVED** [review]

**Review by carletex — COMMENTED** [review]
This is great @mds1! Would have saved a lot time if we had this from the beginning :)

Should we add something about Tailwind? Like when to use utilities, variants. etc. I don't have much experience with it, but maybe someone can chime in and write set some Tailwind standards.

**Comment by apbendi** [comment]
@mds1 maybe you can include a correction to the error you found in my rebasing guide here as well?

File: CONTRIBUTING.md
Comment (mds1) [reply CONTRIBUTING.md]: Updated to frontend in 42a8e96d95f2138fea0243f7438bb8c5d7f4f6a6

File: CONTRIBUTING.md
Comment (mds1) [reply CONTRIBUTING.md]: Done in 42a8e96d95f2138fea0243f7438bb8c5d7f4f6a6

**Comment by mds1** [comment]
> @mds1 maybe you can include a correction to the error you found in my rebasing guide here as well?

@apbendi Done in 42a8e96d95f2138fea0243f7438bb8c5d7f4f6a6

> Should we add something about Tailwind? Like when to use utilities, variants. etc. I don't have much experience with it, but maybe someone can chime in and set some Tailwind standards.

@carletex Great call, added info about that in d72ebbd59a987297e92057dba00adeea57d06275, and also added info about the Vue style guide in 88dac00

**Comment by mk1r** [comment]
@mds1 Looking really good. I would just suggest to add a note somewhere in the rebase section to make it clear not to make any changes to the files when resolving conflicts, unless the changes are directly related to the rebase.

File: CONTRIBUTING.md
Comment (mk1r) [inline CONTRIBUTING.md:145]: I would just suggest to add a note somewhere in the rebase section to make it clear not to make any changes to the files when resolving conflicts, unless the changes are directly related to the rebase.

Comment (mds1) [reply CONTRIBUTING.md:145]: Is there a phrasing you think would be more useful than what we have here: https://github.com/dcgtc/dgrants/pull/167/files#diff-eca12c0a30e25b4b46522ebf89465a03ba72a03f540796c979137931d8f92055R146

It says:

> If conflicts are found, you will have to resolve them now. See here for more info on resolving merge conflicts when rebasing. After successfully rebasing, force push back to your feature branch, but never force push to main:

**Comment by mds1** [comment]
>  Is there a section about setting up hardhat network in metamask somewhere? 

@mk1r There is in `app/README.md` which we link to in the contributing file

> and how do other contributors deal with not having VITE_FLEEK_STORAGE_API_KEY

Good question, @apbendi any idea if there's a free Fleek API key people can register for that would work just as well?

File: CONTRIBUTING.md
Comment (mk1r) [reply CONTRIBUTING.md:145]: hmm, yeah I guess I'll have to figure out better phrasing - this seems a bit confusing to me, since it seems like it's saying to resolve merge conflicts during rebase. I think it would be better if it was explicitly clear that you can't really resolve merge conflicts (things that break other parts of the code because of new commits) until after rebasing is done. 

Even in our Discord chat, the wording is more clear:

> I think a better phrasing would be "no changes outside the merge conflict"
> But fixing things within the conflict is ok

File: CONTRIBUTING.md
Comment (carletex) [inline CONTRIBUTING.md:209]: Style Guide V3 link: https://v3.vuejs.org/style-guide/

Comment (mds1) [reply CONTRIBUTING.md]: Ah nice, done in 5a5d69d961814643f76f99a56755449d62564325

File: CONTRIBUTING.md
Comment (mds1) [reply CONTRIBUTING.md:145]: @apbendi any thoughts here?

Comment (apbendi) [reply CONTRIBUTING.md:145]: Added the following:

> In some circumstances, the process of rebasing and resolving merge conflicts may have broken the feature or functionality you were working on. This tends to occur when large changes were made on `main` to the same section of code being modified in a feature branch. In these cases, simply add additional commit(s) to the head of your feature branch correcting the problems.

Going to merge this PR, but if we feel there is still room for improvement, feel free to open an issue or correcting PR @mds1 @mk1r

**Pull request merged**
